### PR TITLE
feat: Bring multiple data shapes, Step IDs and labels into datamapper

### PR DESCRIPTION
**Pull request by igarashitm**
Fixes: #1135
Fixes: #1137 

I was still debugging though, submitting a PR with WIP flag as I hit a version mismatch issue. Due to the issue now I can't run Syndesis UI with latest AtlasMap UI.

Recently I hit following issue in AtlasMap UI and upgraded the `ng-packagr` to `2.0.0-rc.6`
https://github.com/dherges/ng-packagr/issues/311

Then Syndesis UI fails to build with
```
ERROR in Metadata version mismatch for module /home/tomo/workspace/iPaaS/syndesisio/syndesis/app/ui/node_modules/@atlasmap/atlasmap.data.mapper/atlasmap-atlasmap.data.mapper.d.ts, found version 4, expected 3

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/home/tomo/workspace/iPaaS/syndesisio/syndesis/app/ui/src'
 @ ./src/main.ts 6:29-76
 @ multi ./src/main.ts
```

It seems to be caused by the Angular version mismatch, as AtlasMap is already based on Angular5 while Syndesis UI is still on Angular4.
I tried to upgrade only angular-cli and etc, but couldn't find a happy combination. Apparently we need to update to Angular5.

@deeleman can you help on this?

### CODE REVIEW

File: app/ui/src/app/integration/edit-page/current-flow.service.ts [hunk 1]
```diff
@@ -32,7 +35,10 @@ export class CurrentFlow {
   private _integration: Integration;
   private _loaded = false;
 
-  constructor(private store: IntegrationStore) {
+  constructor(
+    private store: IntegrationStore,
```
Comment (gashcrumb) [inline app/ui/src/app/integration/edit-page/current-flow.service.ts]: Can we rename this to `integrationStore`?  I know the original code was like that but it'd be good to align the instance name to the service class, should be a matter of hitting `F2` in vscode if you use that for your typescript :-)

File: app/ui/src/app/integration/edit-page/current-flow.service.ts [hunk 2]
```diff
@@ -32,7 +35,10 @@ export class CurrentFlow {
...
+    private integrationSupport: IntegrationSupportService
```
Comment (gashcrumb) [inline app/ui/src/app/integration/edit-page/current-flow.service.ts]: Same here, be better to call it `integrationSupportService`

File: app/ui/src/app/integration/edit-page/step-configure/step-configure.component.html [hunk 3]
```diff
@@ -64,11 +64,8 @@
           <div *ngSwitchCase="'mapper'">
             <div class="row">
               <div class="col-md-12"
-                   *ngIf="step && inputDataShape && outputDataShape">
-                <syndesis-data-mapper-host [inputDataShape]="inputDataShape"
-                                           [outputDataShape]="outputDataShape"
-                                           [step]="step"
-                                           [position]="position"></syndesis-data-mapper-host>
+                   *ngIf="step">
+                <syndesis-data-mapper-host [step]="step" [position]="position"></syndesis-data-mapper-host>
```
Comment (gashcrumb) [inline app/ui/src/app/integration/edit-page/step-configure/step-configure.component.html]: probably don't even need to pass the step along TBH, you could get it from the injected `CurrentFlow` service via the position.

File: app/ui/src/app/integration/edit-page/step-configure/step-configure.component.ts [hunk 4]
```diff
@@ -179,11 +138,34 @@ export class IntegrationStepConfigureComponent extends FlowPage implements OnIni
...
+          })
+          .catch(response => this.handleDataShapeError(nextStep, response))
+      })
+      .catch(response => this.handleDataShapeError(prevStep, response))
       .then(() => this.loadFormSetup(step));
   }
 
+  private handleDataShapeError(step: Step, response: any) {
+    this.loading = false;
+    const error = JSON.parse(response['_body']);
```
Comment (gashcrumb) [inline app/ui/src/app/integration/edit-page/step-configure/step-configure.component.ts]: we'll need to double-check what excatly comes back in this error case, I'm not sure that the returned object is a `Response`, so `error` could wind up being undefined.

**Comment by gashcrumb** [comment]
This is all awesome stuff, great work!.  Am not sure on the versioning/error myself, I thought @deeleman had said it should still kinda work but that was awhile ago, my memory only extends back like a couple days at most :-)

File: app/ui/src/app/integration/edit-page/step-configure/step-configure.component.ts (hunk 4 above)
Comment (igarashitm) [reply app/ui/src/app/integration/edit-page/step-configure/step-configure.component.ts]: Added a safeguard against NPE for now, let's see when test the error path.

**Comment by igarashitm** [comment]
Yep, actually it has been working until yesterday. Today suddenly I hit that weird tsickle error in ng-packagr without doing any dependency change... weird world.

**Comment by igarashitm** [comment]
memo: If we can get around the dherges/ng-packagr#311 without updating ng-packagr to 2.0rc, that would be reasonable.

**Comment by deeleman** [comment]
Closing, as superseded by #1367

**Pull request closed without merging**